### PR TITLE
Add new sign in flow

### PR DESCRIPTION
**Pull request by tgolen**
# NOTE
This is not hooked up to any APIs yet, it just walks through the form steps.

Things left to do:
- [x] Call `GetAccountStatus` to see if the user had an account already and if they have access to e.cash
- [x] Call `SetGithubUsername` to save the github username
- [x] Call `CreateLogin` if account already existed
- [x] Call `CreateAccount` if account didn't exist
- [x] See if the 2FA field can be hidden if it's not required (maybe add this to `GetAccountStatus`?)

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/148719
Fixes https://github.com/Expensify/Expensify/issues/148714
Fixes https://github.com/Expensify/Expensify/issues/148716

### Tests
As you go through the testing, verify that form validation works (you can't submit an empty form)

1. Fill out the login form, verify you are taken to the github username form
2. Fill out the github username form, verify you are taken to the password form
3. Fill out the password form, verify you are taken to the resend validation link form
4. Clear out local storage according to [this SO](https://stackoverflow.com/c/expensify/questions/6923/6924#6924)
5. Refresh the page and verify you are back at the first login form

### Screenshots
|image|image
|image|image

cc @chiragsalian @stitesExpensify @shawnborton

### CODE REVIEW

File: src/pages/signin/LoginForm.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            isLoading: false,
+        };
+    }
+
+    /**
+     * Check that all the form fields are valid, then trigger the submit callback
+     */
+    validateAndSubmitForm() {
+        if (!this.state.login.trim()) {
+            this.setState({formError: 'Please fill out all fields'});
```
Comment (stitesExpensify) [inline src/pages/signin/LoginForm.js:26]: Same thing here

File: src/pages/signin/PasswordForm.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            isLoading: false,
+        };
+    }
+
+    /**
+     * Check that all the form fields are valid, then trigger the submit callback
+     */
+    validateAndSubmitForm() {
+        if (!this.state.password.trim() || !this.state.twoFactorAuthCode.trim()) {
+            this.setState({formError: 'Please fill out all fields'});
```
Comment (stitesExpensify) [inline src/pages/signin/PasswordForm.js:35]: copy

File: src/libs/actions/Session.js [hunk 3]
```diff
@@ -112,7 +133,82 @@ function signOut() {
         .catch(error => Onyx.merge(ONYXKEYS.SESSION, {error: error.message}));
 }
 
+/**
+ * Checks the API to see if an account exists for the given login
+ *
+ * @param {String} login
+ */
+function hasAccount(login) {
```
Comment (marcaaron) [inline src/libs/actions/Session.js]: Maybe we could name this `fetchAccount()` instead? `hasAccount` sounds more like a utility method that would return `true` if `account.accountExists`.

File: src/libs/actions/Session.js [hunk 4]
```diff
@@ -72,25 +95,23 @@ function signIn(partnerUserID, partnerUserSecret, twoFactorAuthCode = '', exitTo
...
+                    // If we have an old login for some reason, we should delete it before storing the new details
+                    if (credentials.login) {
                         API.DeleteLogin({
                             partnerUserID: credentials.login,
                             partnerName: CONFIG.EXPENSIFY.PARTNER_NAME,
                             partnerPassword: CONFIG.EXPENSIFY.PARTNER_PASSWORD,
                             doNotRetry: true,
                         })
-                            .catch(error => Onyx.merge(ONYXKEYS.SESSION, {error: error.message}));
+                            .catch(console.debug);
```
[the remark this answers is not in the document]
Comment (tgolen) [reply src/libs/actions/Session.js:106]: I just think this particular error isn't helpful to surface to the user in the UI (they can't do anything about it), so I'm just sending it to the console now.

File: src/libs/actions/Session.js [hunk 5]
```diff
@@ -112,7 +133,82 @@ function signOut() {
...
+                    accountExists: response.accountExists,
+                    canAccessExpensifyCash: response.canAccessExpensifyCash,
+                });
+            }
+        });
+}
+
+/**
+ * Create a login or an account depending on if the user's account they are logging in
+ * with already had an account associated with it or not.
```
Comment (marcaaron) [inline src/libs/actions/Session.js]: Somewhat confused about which two scenarios we are handling here. Maybe this comment can be more specific and also touch on what the difference is between a `login` and an `account` in this context? Anyone familiar with our API will have some ideas about what is meant here. But contributors are likely to be confused.

Separately, I think it might also be clearer if there were separate methods for handling the specific cases. e.g. if the scenarios are either logging in with an account that exists or creating an account w/ a user provided login & password then the latter seems more like a `signUp` than a `signIn`.

File: src/libs/API.js [hunk 6]
```diff
@@ -254,6 +254,26 @@ function CreateChatReport(parameters) {
...
+ * @param {String} parameters.partnerName
+ * @param {String} parameters.partnerPassword
+ * @param {String} parameters.email
+ * @param {String} parameters.password
+ * @param {String} [parameters.githubUsername]
+ * @returns {Promise}
+ */
+function CreateAccount(parameters) {
+    const commandName = 'CreateAccount';
+    requireParameters([
```
[the remark this answers is not in the document]
Comment (tgolen) [reply src/libs/API.js:268]: I got pretty confused about this and I wasn't sure that it was required in every flow, but I think you're right. I can make it required.

File: src/libs/actions/Session.js [hunk 7]
```diff
@@ -72,25 +95,23 @@ function signIn(partnerUserID, partnerUserSecret, twoFactorAuthCode = '', exitTo
...
+                })
+                .catch((error) => {
+                    Onyx.merge(ONYXKEYS.SESSION, {error: error.message});
                 });
-        })
-        .catch((error) => {
-            console.debug('[SIGNIN] Request error', error);
-            Onyx.merge(ONYXKEYS.SESSION, {error: error.message});
-        })
-        .finally(() => Onyx.merge(ONYXKEYS.SESSION, {loading: false}));
```
[the remark this answers is not in the document]
Comment (tgolen) [reply src/libs/actions/Session.js:93]: No, loading is handled by each individual form in its own state. It's no longer stored in Onyx.

File: src/libs/actions/Session.js [hunk 8]
```diff
@@ -112,7 +133,82 @@ function signOut() {
...
+ * @param {String} twoFactorAuthCode
+ * @param {String} exitTo
+ */
+function signIn(password, twoFactorAuthCode, exitTo) {
+    if (account.accountExists) {
+        authenticateAndCreateLogin(password, twoFactorAuthCode, exitTo);
+        return;
+    }
+
+    createAccount(password, twoFactorAuthCode);
```
[the remark this answers is not in the document]
Comment (tgolen) [reply src/libs/actions/Session.js:170]: ah, OK. I think you're right, yep. I'll switch that around

File: src/pages/signin/GithubUsernameForm.js [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+            isLoading: false,
+        };
+    }
+
+    /**
+     * Check that all the form fields are valid, then trigger the submit callback
+     */
+    validateAndSubmitForm() {
+        if (!this.state.githubUsername.trim()) {
+            this.setState({formError: 'Please fill out all fields'});
```
[the remark this answers is not in the document]
Comment (tgolen) [reply src/pages/signin/GithubUsernameForm.js:25]: I can do that, sure.

File: src/pages/signin/LoginForm.js [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+            return;
+        }
+
+        this.setState({
+            formError: null,
+            isLoading: true,
```
Comment (marcaaron) [inline src/pages/signin/LoginForm.js:32]: NAB, I'd expect this to maybe appear as a `prop` set by the API command as a best practice. If something fails and we do not redirect then this will never be set back to `false`.

File: src/pages/signin/GithubUsernameForm.js [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+            return;
+        }
+
+        this.setState({
+            formError: null,
+            isLoading: true,
```
Comment (marcaaron) [inline src/pages/signin/GithubUsernameForm.js:31]: Same comment here. The loading state only gets set one way since we are banking on a redirect and it's not clear what if anything would unset it.

File: src/pages/signin/LoginForm.js [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+        });
+
+        // Check if this login has an account associated with it or not
+        hasAccount(this.state.login);
+    }
+
+    render() {
+        return (
+            <>
+                <View style={[styles.mb4]}>
```
Comment (marcaaron) [inline src/pages/signin/LoginForm.js:42]: NAB, this form component could maybe be DRY-d up into a component like `SignInPanel` or something

File: src/pages/signin/ResendValidationForm.js [hunk 13]
```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import {Text, View} from 'react-native';
+import styles from '../../styles/styles';
+import SubmitButton from './SubmitButton';
+import {resendValidationLink} from '../../libs/actions/Session';
+
+class ResendValidationForm extends React.Component {
+    constructor() {
+        super();
```
Comment (marcaaron) [inline src/pages/signin/ResendValidationForm.js:9]: Should this and other components be... 

```js
constructor(props) {
    super(props);
    ...
}
```

File: src/pages/signin/ResendValidationForm.js [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+            isLoading: false,
+        };
+    }
+
+    /**
+     * Check that all the form fields are valid, then trigger the submit callback
+     */
+    validateAndSubmitForm() {
+        this.setState({
+            formSuccess: 'Link has been resent',
```
Comment (marcaaron) [inline src/pages/signin/ResendValidationForm.js:24]: `re-sent`

File: src/pages/signin/SignInPage.js [hunk 15]
```diff
@@ -0,0 +1,158 @@
...
+    }),
+};
+
+const defaultProps = {
+    account: {},
+    session: null,
+    credentials: {},
+};
+
+class App extends Component {
```
Comment (marcaaron) [inline src/pages/signin/SignInPage.js]: Can this be `SignInPage` instead of `App` ?

**Comment by tgolen** [comment]
Updated to fix conflicts and address comment. Still hard to tell if the logic is all correct until it can be tested. Once it can be fully tested then I'll bring this out of WIP.

File: src/pages/signin/SignInPage.js [hunk 16]
```diff
@@ -0,0 +1,158 @@
...
+        // Show the resend validation link form if
+        // - A login has been entered
+        // - AND a GitHub username has been entered OR they already have access to this app
+        // - AND an account did not exist for that login
+        const showResendValidationLinkForm = this.props.credentials.login
+            && (
+                this.props.credentials.githubUsername
+                || this.props.account.canAccessExpensifyCash
+            )
+            && !this.props.account.accountExists;
```
Comment (marcaaron) [inline src/pages/signin/SignInPage.js:112]: NAB, this stuff could maybe be abstracted into a helper that returns the correct form to display and then we do or something similar...

```js
viewName === CONST.SIGN_IN.LOGIN && <LoginForm/>
``` 
Another question I have whether we could use routes like `/login` or `/github-name`? Rather than show a different view depending on the conditions of these various props it seems like we could just explicitly redirect to these views when some conditions are met or following the completion of some other action.

More or less just food for thought though. I think this is OK for now.

File: src/styles/styles.js [hunk 17]
```diff
@@ -37,17 +37,21 @@ const styles = {
...
-        marginTop: 20,
-    },
-
     mt1: {
         marginTop: 4,
     },
-
+    mt2: {
+        marginTop: 8,
+    },
```
Comment (marcaaron) [inline src/styles/styles.js:45]: NAB, but we should add some consistency to the values used for these. If we are incrementing these helpers by `4` that's cool but we should follow up and do this everywhere.

File: src/pages/signin/ResendValidationForm.js [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+        });
+
+        resendValidationLink();
+
+        setTimeout(() => {
+            this.setState({formSuccess: ''});
+        }, 5000);
```
Comment (marcaaron) [inline src/pages/signin/ResendValidationForm.js:31]: Why is this necessary?

File: src/pages/signin/PasswordForm.js [hunk 19]
```diff
@@ -0,0 +1,125 @@
...
+        // Whether or not two factor authentication is required
+        requiresTwoFactorAuth: PropTypes.bool,
+    }),
+};
+
+const defaultProps = {
+    account: {},
+};
+
+class PasswordForm extends React.Component {
```
Comment (marcaaron) [inline src/pages/signin/PasswordForm.js:37]: NAB, it wouldn't hurt to add brief descriptions to each of these views to explain in which situations they are used.

**Review by marcaaron — COMMENTED** [review]
Just leaving some initial thoughts for now. Some of this stuff can likely be addressed in a follow-up.

File: src/pages/signin/SignInPage.js [hunk 20]
```diff
@@ -0,0 +1,158 @@
...
+
+        return (
+            <>
+                <CustomStatusBar />
+                <SafeAreaView style={[styles.signInPage]}>
+                    <View style={[styles.signInPageInner]} accessibilityRole="form">
```
Comment (AndrewGable) [inline src/pages/signin/SignInPage.js:118]: I don't think iOS likes this `accessibilityRole`:

Screen Shot 2020-12-21 at 11 35 29 AM

**Comment by AndrewGable** [comment]
FYI - Login isn't saved when I click "Change Expensify login" to go back, should it be?

File: src/pages/signin/LoginForm.js (hunk 10 above)
Comment (tgolen) [reply src/pages/signin/LoginForm.js:32]: We used to have `session.loading` stored in Onyx, but I thought it was better for each form to control its own state. I don't feel too strongly about it though and could really go either way.

File: src/pages/signin/SignInPage.js (hunk 16 above)
Comment (tgolen) [reply src/pages/signin/SignInPage.js:112]: That's an interesting thought about using routers. That could be a good way to organize this, but I'll save that for later. A possible downside is that each view would essentially need to control when it can/can't be displayed (in case someone uses the browser navigation to go between steps). That would make it a little bit more complex in the end I think.

File: src/styles/styles.js (hunk 17 above)
Comment (tgolen) [reply src/styles/styles.js:45]: We do mostly do this everywhere already, but yes, we should be consistent and I fixed a bit of the inconsistency here.

File: src/pages/signin/ResendValidationForm.js (hunk 18 above)
Comment (tgolen) [reply src/pages/signin/ResendValidationForm.js:31]: I just didn't like the success message staying there forever. I think there is a diminishing return for a success message remaining visible

File: src/pages/signin/SignInPage.js (hunk 20 above)
Comment (tgolen) [reply src/pages/signin/SignInPage.js:118]: Thanks, Andrew. I was going to look at mobile once all the APIs were working, but you beat me to it! I'll just remove it for now since it only results in a browser warning. Sounds like we'll need to do a no-op for this on mobile.

**Comment by tgolen** [comment]
> FYI - Login isn't saved when I click "Change Expensify login" to go back, should it be?

I don't think it should be saved, no. It doesn't make sense that if you wanted to change your login that you would want the old value to be there (since you're explicitly trying to change it).

@marcaaron Thanks for the review! I've addressed some of those. Some of the NABs I'm not going to do in this PR and we can do later if we want. Some of your comments were on outdated code as well, so some of the confusion might have been cleared up already.

**Comment by stitesExpensify** [comment]
Is it intentional that the web version kinda looks like a mobile version?

**Comment by quinthar** [comment]
Hm, ya ideally we'd have a more responsive design for desktop.

**Comment by tgolen** [comment]
That is sort of intentional... we can do better though. I'll have another PR to try and improve the desktop design.

This is all tested and ready to go, so finally removing WIP

**Comment by stitesExpensify** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/libs/actions/Session.js [hunk 21]
```diff
@@ -33,37 +32,102 @@ function setSuccessfulSignInData(data, exitTo) {
...
+
+/**
+ * Sign the user into the application. This will first authenticate their account
+ * then it will create a temporary login for them which is used when re-authenticating
+ * after an authToken expires.
+ *
+ * @param {String} password
  * @param {String} [twoFactorAuthCode]
- * @param {String} [exitTo]
+ * @param {String} exitTo
```
Comment (chiragsalian) [inline src/libs/actions/Session.js:108]: Required params should be before optional ones.

**Review by chiragsalian — APPROVED** [review]
Did a quick review. Merging early as requested since we want to get this out the door soon.

File: src/libs/actions/Session.js (hunk 21 above)
Comment (tgolen) [reply src/libs/actions/Session.js:108]: I'll fix that in my next PR

File: src/pages/signin/LoginForm.js (hunk 10 above)
Comment (marcaaron) [reply src/pages/signin/LoginForm.js:51]: It would be great if we could aim for consistency here. Considering that loading states usually are tied to some kind of async action it could make sense to hook these into props passed via `Onyx`. 

We aren't going to be directly handling the result of any action's API request because `Onyx`. And I'm not really sure what the best solution would be but having one way loading states that are not longer loading because you unmounted a view doesn't seem right.

File: src/pages/signin/ResendValidationForm.js (hunk 18 above)
Comment (marcaaron) [reply src/pages/signin/ResendValidationForm.js:31]: Didn't quite understand your meaning here. Why does it matter if it disappears?

File: src/pages/signin/SignInPage.js [hunk 22]
```diff
@@ -0,0 +1,157 @@
...
+const defaultProps = {
+    account: {},
+    session: null,
+    credentials: {},
+};
+
+class SignInPage extends Component {
+    componentDidMount() {
+        // Always reset the unread counter to zero on this page
+        updateUnread(0);
```
Comment (marcaaron) [inline src/pages/signin/SignInPage.js:60]: This feels like it should happen somewhere else like the main page with all the routes on it.

File: src/pages/signin/SignInPage.js (hunk 16 above)
Comment (marcaaron) [reply src/pages/signin/SignInPage.js:108]: >A possible downside is that each view would essentially need to control when it can/can't be displayed

Not really sure what you mean maybe you can explain it another way? Views don't control when they can or can't be displayed you just display them or you do not display them based on a `Route` component.

File: src/libs/actions/Session.js [hunk 23]
```diff
@@ -73,44 +137,60 @@ function signIn(partnerUserID, partnerUserSecret, twoFactorAuthCode = '', exitTo
 
                     setSuccessfulSignInData(createLoginResponse, exitTo);
 
-                    if (credentials && credentials.login) {
-                        // If we have an old login for some reason, we should delete it before storing the new details
+                    // If we have an old login for some reason, we should delete it before storing the new details
+                    if (credentials.login) {
```
Comment (marcaaron) [inline src/libs/actions/Session.js:141]: `credentials.login` got populated after `GetAccountStatus`. I'm guessing that is not the login we want to be deleting here.

File: src/libs/actions/Session.js [hunk 24]
```diff
@@ -73,44 +137,60 @@ function signIn(partnerUserID, partnerUserSecret, twoFactorAuthCode = '', exitTo
...
                             partnerName: CONFIG.EXPENSIFY.PARTNER_NAME,
                             partnerPassword: CONFIG.EXPENSIFY.PARTNER_PASSWORD,
                             doNotRetry: true,
                         })
-                            .catch(error => Onyx.merge(ONYXKEYS.SESSION, {error: error.message}));
+                            .catch(console.debug);
                     }
 
-                    Onyx.merge(ONYXKEYS.CREDENTIALS, {login, password});
+                    Onyx.merge(ONYXKEYS.CREDENTIALS, {password});
```
Comment (marcaaron) [inline src/libs/actions/Session.js:151]: I _think_ removing `login` from here means that we are no longer saving the generated login we created and are also just saving the original password because of the variable name was changed. But not sure if there was some other plan at work

File: src/pages/signin/LoginForm.js (hunk 10 above)
Comment (tgolen) [reply src/pages/signin/LoginForm.js:51]: Yeah, I agree that this can be more consistent. Do you think that there should be one loading state to rule the entire app (ie. `app.isLoading`) or should there be a loading state for every view (ie. `loginForm.isLoading`, `passwordForm.isLoading`, etc.)? I kind of think the one loading state would be OK for now, and it could eventually turn into the equivalent of `UIUtils.Loading.show()` and `UIUtils.Loading.hide()` from web-e which seems to work pretty well.

Comment (marcaaron) [reply src/pages/signin/LoginForm.js:51]: I was thinking the loading state would be tied to the data model that you are waiting to load...

So e.g. when entering your `email` step we would: 
1. Set `account.loading` 
2. Call `GetAccountStatus` 
3. Set `account.loading` to false once the data is available

A simpler approach could be taken with error states.

File: src/pages/signin/SignInPage.js (hunk 16 above)
Comment (tgolen) [reply src/pages/signin/SignInPage.js:108]: I mean that this logic that you commented on in the PR still needs to exist somewhere. It can't be entirely controlled by the Router. I'll give an example:

- A route for `/loginForm` is added to display `LoginForm`
- The `LoginForm` can only be displayed when `!credentials.login`
- A user enters their email into `LoginForm` and it's saved to `credentials.login`
- If a user manually navigates to `/loginForm` (by typing into the address bar, or using the back buttons on the browser), then there has to be some logic (I assume this would live inside the view) that would say "The router says I should show the `LoginForm`, but the user has already entered their login, so I should probably not show the `LoginForm` again.

File: src/pages/signin/ResendValidationForm.js (hunk 18 above)
Comment (tgolen) [reply src/pages/signin/ResendValidationForm.js:31]: > Why does it matter if it disappears?

I'll turn the question back around (since I also don't understand why you asked the original question). Why does this matter if it disappears after 5 seconds?

File: src/pages/signin/LoginForm.js (hunk 10 above)
Comment (tgolen) [reply src/pages/signin/LoginForm.js:51]: In that case, then don't you think it's a waste of logic to set the loading state to `false,` knowing that the view is immediately going to be unmounted? In other words, there would be no UI to show that `account.loading` is false again, so it's just wasted data sitting in Ion with no views bound to it.

File: src/pages/signin/SignInPage.js (hunk 16 above)
Comment (marcaaron) [reply src/pages/signin/SignInPage.js:108]: Ah ok thanks for the explanation! That's interesting. And I get what you are saying now. 

So I don't think we have to figure this out immediately. But this might come up again. Questions like...

- When should we create a new route for a view? (Probably, if it needs to be "deep linked" to or should respond to native navigation actions like back button etc)
- When we are _not_ creating a new route for a view or series of views - what does that look like? Is there some kind of best practice we can use to create a flow that has a series of steps. Does it bypass the usual navigation? What is it called? I think this is maybe a kind of "Stepper" navigation, but without feedback.

Comment (tgolen) [reply src/pages/signin/SignInPage.js:108]: > When should we create a new route for a view?

I think the criteria you started with is great.

> When we are not creating a new route for a view or series of views - what does that look like?

Yeah, it would be nice to make this consistent, though I think there is a risk of over-engineering this since we only have one use-case right now, and we don't have a clear idea of what other use cases might be. This could be something that we keep in mind for a refactoring the next time we notice a view has this multi-step approach to it.

File: src/pages/signin/ResendValidationForm.js (hunk 18 above)
Comment (marcaaron) [reply src/pages/signin/ResendValidationForm.js:31]: I asked this because I didn't understand _why_ we are doing it. And it would be nice to have a comment in the code to explain why.

File: src/pages/signin/LoginForm.js (hunk 10 above)
Comment (marcaaron) [reply src/pages/signin/LoginForm.js:51]: I can see that perspective, but I think it's not a waste since we maybe can't or should not assume that the view will immediately be unmounted and that nothing will ever go wrong with this step? 

We also are creating another way to do something when perhaps it would be better to standardize on one single way. 

Also small data like `loading: false` is very cheap. If that was a concern we could go from `loading: true` to no loading flag at all. But I don't think that's a problem we'd ever have.

Comment (tgolen) [reply src/pages/signin/LoginForm.js:51]: OK, thanks. As I was thinking about how this multi-step view could be refactored at some point, I was actually thinking that it would be good to tie the loading setting to the entire set of steps. So if we had several multi-step views, each of those flows would have its own loading setting in Onyx. Would you be OK with that? I think it's a best of both worlds where we don't have everything using a global setting, but also each individual form isn't using it's own personal setting either.

Comment (marcaaron) [reply src/pages/signin/LoginForm.js:51]: Sure that sounds ok for now. 

I do think it would be beneficial to maybe start a wider convo about how handling loading and error states in general should work in the app. My main concern would be having a bunch of different ways to do it. It's an interesting topic, but one that we haven't had to deal with outside of the sign in flow (I suspect this will change).

Comment (tgolen) [reply src/pages/signin/LoginForm.js:51]: Cool, I created an issue to get to this. Thanks!

Comment (tgolen) [reply src/pages/signin/LoginForm.js:51]: (I meant to refactor `isLoading`... I didn't create an issue to have that bigger discussion)

Comment (marcaaron) [reply src/pages/signin/LoginForm.js:51]: Sounds good! We got some time to hash that stuff out. 

FWIW, there was a decent conversation about it around the time when this was created --> https://github.com/Expensify/Expensify.cash/pull/556/files so my suggestion was based on that precedent.
